k4FWCore/components/Writer.cpp

```diff
@@ -233,12 +235,12 @@ class Writer final : public Gaudi::Functional::Consumer<void(const EventContext&
         return;
       }
       // We still have to delete the AnyDataWrapper to avoid a leak
-      auto storePtr = dynamic_cast<AnyDataWrapper<std::unique_ptr<podio::CollectionBase>>*>(storeCollection);
-      storePtr->getData().release();
+      const auto storePtr = dynamic_cast<AnyDataWrapper<std::unique_ptr<podio::CollectionBase>>*>(storeCollection);
+      [[maybe_unused]] auto releasedPtr = storePtr->getData().release();
```

Comment:
The `maybe_unused` silences a warning that we don't have enabled (maybe we should) about not using the result of a `release()` which in most cases will be a leak since that won't be destroyed but in this case it is. It shouldn't change anything after compiling. I can add a comment about this.

Reply:
Thank you for the explanation! Please add a comment.

Reply:
Done